update links and wording of `redpanda --help` output

andrewhsu opened the pull request:
## Cover letter

To adjust for company rename and clean up the phrasing.

Also reordered the list in the sequence that would be generally recommended for finding answers to questions--putting docs link first.

## Release notes

* none

coral-waters commented:
Is it standard to use informal language and lowercase, etc? Or can I clean it up?

coral-waters commented:
Ok. I made some suggestions. You can apply them at your discretion.

andrewhsu commented:
I've accepted all of @coral-waters 's suggestions and squashed them into 1 commit. I also rebased with [dev](https://github.com/redpanda-data/redpanda/tree/dev) branch.

dotnwat commented:
Unrelated failure. Linked to issue.
